src/list/manage/index.js

```diff
@@ -15,22 +15,29 @@ import reducer from "./reducers";
 import initializeMessagePorts from "../message-ports";
 import * as telemetry from "../../telemetry";
 import telemetryLogger from "./telemetry";
+import { saveSort, loadSort } from "./sort-middleware";
 
-const store = createStore(reducer, undefined, applyMiddleware(
-  thunk, telemetryLogger
-));
-store.dispatch(listItems());
-initializeMessagePorts(store);
-store.dispatch(getProfile());
+let store;
+(async () => {
```

Comment:
Hmm, this *should* be near-immediate and not require any kind of visual loading indicator. But, wrapping the entire initialization in an async block waiting on the sort loading seems like it could be trouble.

Could you initialize the app, *then* start the async load and dispatch an action to update the state? (I guess that might cause a flash of content shifting around.) Also, is there a chance that an exception could get thrown here?

Reply:
More good questions! Re-sorting the list is going to be a highly visible change, so it seemed better to force it to wait a beat to hit localstorage.

There is indeed exception handling to deal with, but that happens in sort-middleware.js, in which case the `loadSort` return value is falsy, and we'll wind up going with the default sort.